### PR TITLE
opera.xml

### PR DESCRIPTION
**Pull request by Tobias-B-Besemer**
I got the following error in the console of BB v2.1.876:
2019-03-13 02_21_31-Administrator_ C__Windows_system32_cmd exe - bleachbit_console exe  --gui --debu

I had a look on the cleaner and guessed that all ".DAT" are files...

I'm just not sure, if `search="file"` supports wildcards in the path...
...if not, I think this is inconsequent and should be changed.

### CODE REVIEW

**Comment by Tobias-B-Besemer** [comment]
I use this lines in my new visual_studio.xml...
`<action command="delete" search="file" path="%AppData%\Microsoft\VisualStudio\*\ActivityLog.Setup.xml"/>`
`<action command="delete" search="file" path="%AppData%\Microsoft\VisualStudio\*\ActivityLog.xsl"/>`
I would prefer if it works...

File: cleaners/opera.xml [hunk 1]
```diff
@@ -78,8 +86,8 @@ Notes about Opera
     <!-- below are entries valid before Opera 15 -->
     <!-- example: C:\Documents and Settings\user\Application Data\Opera\Opera\cookies.txt -->
     <!-- example: C:\Documents and Settings\user\Application Data\Opera\Opera 11.00 beta\cookies.txt -->
-    <action command="delete" search="glob" path="$appdata\Opera\Opera*\cookies4.dat"/>
-    <action command="delete" search="glob" path="$appdata\Opera\Opera*\profile\cookies4.dat"/>
+    <action command="delete" search="file" path="$appdata\Opera\Opera*\cookies4.dat"/>
+    <action command="delete" search="file" path="$appdata\Opera\Opera*\profile\cookies4.dat"/>
```
Comment (az0) [inline cleaners/opera.xml]: When you put a wildcard in `path`, then set `search="glob"` 

I don't understand the reason for this change.

File: cleaners/opera.xml [hunk 2]
```diff
@@ -119,19 +127,19 @@ Notes about Opera
...
     <!-- global.dat is in Opera 9, global_history.dat is in Opera 10 -->
     <action command="delete" search="file" path="$appdata\Opera\Opera\profile\global.dat"/>
     <action command="delete" search="file" path="$appdata\Opera\Opera\profile\typed_history.xml"/>
     <action command="delete" search="file" path="$appdata\Opera\Opera\profile\vlink4.dat"/>
     <action command="delete" search="file" path="$localappdata\Opera\Opera\profile\vps\????\md.dat"/>
-    <action command="delete" search="glob" path="$appdata\Opera\Opera*\global_history.dat"/>
+    <action command="delete" search="file" path="$appdata\Opera\Opera*\global_history.dat"/>
     <action command="delete" search="glob" path="$appdata\Opera\Opera*\typed_history.xml"/>
-    <action command="delete" search="glob" path="$appdata\Opera\Opera*\vlink4.dat"/>
+    <action command="delete" search="file" path="$appdata\Opera\Opera*\vlink4.dat"/>
```
Comment (az0) [inline cleaners/opera.xml]: Same comments apply to these changes as the above about `file` vs `glob`

**Comment by az0** [comment]
`search="file"` is literal and does not support wildcards

`search="glob"' supports wildcards

File: cleaners/opera.xml (hunk 1 above)
Comment (Tobias-B-Besemer) [reply cleaners/opera.xml]: Yeah, was BS! ^^

**Comment by az0** [comment]
Target to 2.4 for testing

File: cleaners/opera.xml (hunk 2 above)
Comment (Tobias-B-Besemer) [reply cleaners/opera.xml]: Yes!

**Comment by az0** [comment]
I delayed the milestone to focus on Python 3, so BleachBit is not get removed from Debian

Also there are merge conflicts

File: cleaners/opera.xml [hunk 3]
```diff
@@ -29,13 +37,20 @@ Notes about Opera
   <running type="exe" os="linux">opera</running>
   <running type="exe" os="windows">opera.exe</running>
   <var name="base">
-    <value>%UserProfile%\AppData\Roaming\Opera Software\Opera Stable</value>
-    <value>$XDG_CONFIG_HOME/opera</value>
+    <!-- Lets make it "universal"... Was:
+    <value os="windows">%AppData%\Opera Software\Opera Stable</value>
+    Is now: -->
+    <value os="windows">%AppData%\Opera Software\Opera*</value>
```
Comment (az0) [inline cleaners/opera.xml]: I tested today on Windows 10 with Opera beta (Opera version 68). It made a directory `Opera Software\Opera Next`.

When the username was ASCII, this worked. When the username had Unicode, this failed unless I edited the cleaner to replace the glob with the exact name.

**Comment by az0** [comment]
There were merge conflicts, so I resolved them

**Pull request merged**